src/app/app.component.ts

```diff
@@ -20,51 +24,91 @@ export class AppComponent implements OnInit {
   private msg;
   private boundingBoxes = [[]];
   private imgName = '';
+  private imgData;
+  private detectedBoundingBoxes;
+  private detectedLabels;
+  private allDetectedProps;
+  public loading = false;
 
-  ngOnInit() {
-    $('#imageCanvas').mousedown(e => this.handleMouseDown(e));
-    $('#imageCanvas').mouseup(e => this.handleMouseUp(e));
-  }
-
-  private getImages() {
-    this.buildings = document.getElementById('imageImport')['files'];
+  onChange(event) {
+    this.buildings = event.srcElement.files;
     this.imgAmount = Object.keys(this.buildings).length;
-    this.processCanvas();
     this.showUploader = false;
+    this.index = 0;
+    this.showBBoxes();
+  }
+
+  showBBoxes() {
+    const postData = {};
+    this.loading = true;
+    this._service.postWithFile('http://127.0.0.1:5000/', postData, this.buildings).then(result => {
+      this.allDetectedProps = result;
+      this.detectedBoundingBoxes = this.allDetectedProps[this.index]['bboxes'];
+      this.detectedLabels = this.allDetectedProps[this.index]['labels'];
+      this.processCanvas(this.detectedBoundingBoxes, this.detectedLabels);
+    });
   }
 
   private nextImage() {
-    this.saveToCsv();
     this.index++;
     if (this.index >= this.imgAmount) {
       this.showUploader = true;
       this.msg = 'All images have been tagged. Upload more!';
       this.index = 0;
     } else {
-      this.processCanvas();
+      this.detectedBoundingBoxes = this.allDetectedProps[this.index]['bboxes'];
+      this.detectedLabels = this.allDetectedProps[this.index]['labels'];
+      this.processCanvas(this.detectedBoundingBoxes, this.detectedLabels);
     }
   }
 
-  private processCanvas() {
+  private processCanvas(detectedBoundingBoxes, detectedLabels) {
     const reader = new FileReader();
-
     reader.onload = function (e) {
       const canvas: any = document.getElementById('imageCanvas');
       const ctx = canvas.getContext('2d');
       const img: any = new Image();
+      const labelMapping = {
+        0: 'other',
+        1: 'berlinerdom',
+        2: 'brandenburgertor',
+        3: 'fernsehturm',
+        4: 'funkturm',
+        5: 'reichstag',
+        6: 'rotesrathaus',
+        7: 'siegessaeule',
+        8: 'none'
+      };
       img.onload = function () {
         canvas.width = img.width;
         canvas.height = img.height;
         ctx.drawImage(img, 0, 0);
+        if (detectedBoundingBoxes.length !== 0) {
+          detectedBoundingBoxes[0].forEach((item, index) => {
+            ctx.beginPath();
+            const x = detectedBoundingBoxes[0][index][0];
+            const y = detectedBoundingBoxes[0][index][1];
```

Comment:
It would make a little bit more sense to exchange the x and y here (like y = [...][index][0] and x = [...][index][1], because the method below takes (x, y, w, h), but that's just minor.